qiskit_experiments/library/characterization/analysis/drag_analysis.py

```diff
@@ -140,35 +150,32 @@ def _generate_fit_guesses(
         x_data = self._data("series-0").x
         min_beta, max_beta = min(x_data), max(x_data)
 
-        freqs_guesses = {}
-        for i in range(3):
-            curve_data = self._data(f"series-{i}")
-            freqs_guesses[f"freq{i}"] = curve.guess.frequency(curve_data.x, curve_data.y)
-        user_opt.p0.set_if_empty(**freqs_guesses)
+        # Use the highest-frequency curve to estimate the oscillation frequency.
+        curve_data = self._data(f"series-2")
+        freqs_guess = curve.guess.frequency(curve_data.x, curve_data.y) / self.options.reps3
+        user_opt.p0.set_if_empty(freq=freqs_guess)
 
         max_abs_y, _ = curve.guess.max_height(self._data().y, absolute=True)
-        freq_bound = max(10 / user_opt.p0["freq0"], max(x_data))
+        freq_bound = max(10 / user_opt.p0["freq"], max(x_data))
 
         user_opt.bounds.set_if_empty(
             amp=(-2 * max_abs_y, 0),
-            freq0=(0, np.inf),
-            freq1=(0, np.inf),
-            freq2=(0, np.inf),
+            freq=(0, np.inf),
             beta=(-freq_bound, freq_bound),
             base=(-max_abs_y, max_abs_y),
         )
-        user_opt.p0.set_if_empty(base=0.5)
+        user_opt.p0.set_if_empty(base=(user_opt.p0["amp"] or max_abs_y) / 2)
 
         # Drag curves can sometimes be very flat, i.e. averages of y-data
         # and min-max do not always make good initial guesses. We therefore add
         # 0.5 to the initial guesses. Note that we also set amp=-0.5 because the cosine function
         # becomes +1 at zero phase, i.e. optimal beta, in which y data should become zero
         # in discriminated measurement level.
         options = []
-        for amp_guess in (0.5, -0.5):
+        for amp_factor in (-1, -0.5, -0.25):
```

Comment:
Should we be using `np.ptp(y)` instead of max y? Typically the amp should be 0.5 times the peak to peak. I can understand guessing a larger number in case we are not measuring a full period. Guessing a smaller number doesn't seem necessary, though I could see doing it if there is a large baseline and a small amp and you are guessing based on max y instead of ptp y.

Reply:
For cases where we do not have a full oscillation guessing based on the peak to peak of the y data might not work well. This is why I was considering these three cases.

Reply:
Sounds good see here: [946db8c](https://github.com/Qiskit/qiskit-experiments/pull/732/commits/946db8c7c586d61903ff4471ae93f6f879b05990). Level 2 is prefered but level 1 should be possible. I changed to using `ptp` and `(max(y) - min(y)) / 2`.